tests/test_run_reference_vmm.py

```diff
@@ -102,6 +96,52 @@ def expect_string(vmm_process, expected_string):
         if expected_string in nextline.decode():
             break
 
+def expect_vcpus(vmm_process, expected_vcpus):
+    # /proc/cpuinfo displays info about each vCPU
+    vmm_process.stdin.write(b'cat /proc/cpuinfo\n')
+    vmm_process.stdin.flush()
+
+    # Poll stdout to find the 'siblings' field, which displays the total number
+    # of vCPUs, and compare it against the expected number of vCPUs
+    # format of the field: 'siblings  : num'
+    while True:
+        nextline = vmm_process.stdout.readline().decode()
+        if "siblings" in nextline:
+            # collapse all whitespace (e.g., 'siblings:num')
+            nextline = "".join(nextline.split())
+            # extract the number of vCPUs
+            actual_vcpus = int(nextline.split(':')[1])
+            break
+
+    assert actual_vcpus == expected_vcpus, \
+            "Expected {}, found {} vCPUs".format(expected_vcpus, actual_vcpus)
+
+def expect_mem(vmm_process, expected_mem_mib):
+    expected_mem_kib = expected_mem_mib * 1000
+    expected_mem_kb = expected_mem_kib * 1024/1000
+
+    # /proc/meminfo displays info about memory
+    vmm_process.stdin.write(b'cat /proc/meminfo\n')
+    vmm_process.stdin.flush()
+
+    # The MemTotal field displays the total amount of memory in kB
+    # Poll stdout to find the 'MemTotal' field and compare against
+    # the expected amount of memory
+    while True:
+        nextline = vmm_process.stdout.readline().decode()
+        if "MemTotal" in nextline:
+            break
+
+    # Remove whitespace and extract the memory value
+    # Expected format is 'MemTotal: [value] kB'
+    mem_total = nextline.split()[1]
+
+    # Expect the difference between the expected and actual memory
+    # to be less than 10%
```

Comment:
On a second thought, 10% seems like a lot. Would a fixed diff in KB work as well?